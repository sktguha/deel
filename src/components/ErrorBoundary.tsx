import { Component } from "react";

export default class ErrorBoundary extends Component {
    constructor(props: any) {
      super(props);
      this.state = { hasError: false };
    }
  
    static getDerivedStateFromError() {
      return { hasError: true };
    }
  
    render() {
      if ((this.state as {hasError: boolean}).hasError) {
        return <h2>Something went wrong in Autocomplete component. please try again</h2>;
      }
  
      return (this.props as {children: any}).children; 
    }
  }