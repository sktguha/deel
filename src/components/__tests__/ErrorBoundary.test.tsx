import React from 'react';
import ErrorBoundary from "../ErrorBoundary"
import { render , screen } from '@testing-library/react';


test("ErrorBoundary base",()=>{
    const { container } = render(<ErrorBoundary />)
    expect(container).toMatchSnapshot();
})

test("ErrorBoundary error component",()=>{
  const Component = ()=>{
    throw new Error("test error");
    return <div>Test Component</div>
  }
  const { container } = render(<ErrorBoundary><Component /></ErrorBoundary>)
  expect(container).toMatchSnapshot();
})