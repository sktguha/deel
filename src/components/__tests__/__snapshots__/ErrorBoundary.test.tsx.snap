// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary base 1`] = `<div />`;

exports[`ErrorBoundary error component 1`] = `
<div>
  <h2>
    Something went wrong in Autocomplete component. please try again
  </h2>
</div>
`;
